core/interval/int.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package interval
+
+import (
+	"sort"
+)
+
+type IntegerInterval struct {
+	Lower int
+	Upper int
```

Comment:
Need comments on weather these are inclusive or not (look like they are?).

Reply:
Yep, inclusive